### PR TITLE
Fixes #1573

### LINKED ISSUE #1573: App crashes on pressing back button on review loan page
**Summary:**
On Review loan page if back button from toolbar or back button is pressed app crashes. Problem is with Purpose of Loan spinner

**Expected behaviour:**
It should load all selected and entered values on Submit loan page

**Device and Android version:**
Lenovo Tab7 
Android 7.0
Stock ROM

Screenshot_20201126-235513
Screenshot_20201126-235526

### PR DESCRIPTION
**Pull request by PatelVatsalB21**
Fixes #1573

### CODE REVIEW

**Comment by ReCodee** [comment]
@PatelVatsalB21 This PR doesn't follow the contribution guidelines mentioned in the documentation. I recommend going through `README.md` properly and make sure to follow the pull request template.

**Comment by PatelVatsalB21** [comment]
Yes sure but I am new to it and I was having problem that new commit was going in old pr, so sorry about that. Can you help me with that problem?

**Comment by ReCodee** [comment]
> Yes sure but I am new to it and I was having problem that new commit was going in old pr, so sorry about that. Can you help me with that problem?

Sure, Could you post your problem on gitter community channel in detail.

**Comment by PatelVatsalB21** [comment]
Thanks 😊

File: app/src/main/java/org/mifos/mobile/utils/Network.kt [hunk 1]
```diff
@@ -15,9 +15,13 @@ object Network {
      * @param context
      * @return
      */
-    private fun getNetworkInfo(context: Context): NetworkInfo {
-        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        return cm.activeNetworkInfo
+    private fun getNetworkInfo(context: Context): NetworkInfo? {
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobile/utils/Network.kt:18]: Are these changes needed?

**Review by miPlodder — CHANGES_REQUESTED** [review]
Please format your changes before raising PR and don't push unwanted commits, stash them before raising a PR

File: app/src/main/java/org/mifos/mobile/utils/Network.kt (hunk 1 above)
Comment (PatelVatsalB21) [reply app/src/main/java/org/mifos/mobile/utils/Network.kt:18]: yes as previously function required non null active network info but if not connected to internet it will return null and that was error.

Whole function can be optimised as

private fun getNetworkInfo(context: Context): NetworkInfo? {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return cm.activeNetworkInfo
    }
I will change them asap.

Comment (PatelVatsalB21) [reply app/src/main/java/org/mifos/mobile/utils/Network.kt:18]: Sorry for those commits, initially i did only one but while working on other issue it got commited here automatically. I have recently started using git so i was unaware of such issue. Is there anything I can do now to merge all commits?

**Comment by miPlodder** [comment]
No Issues,
Close this and recreate a new PR, start using different branches for
different issues.
You can refer my Gist on Git -
https://gist.github.com/miPlodder/f964112753b560960b51bb0c584d4870

**Comment by PatelVatsalB21** [comment]
Thank you for your guidance
